### ISSUE TITLE
Rancher rancher-images.txt includes docker.io/busybox which can not be downloaded due to "too many requests"

### ISSUE THREAD

**Issue by Martin-Weiss** [rancher/rancher#35587]
**Rancher Server Setup**

AIRGAPPED

- Rancher version: 2.5.11 and 2.6.2
- Installation option (Docker install/Helm Chart): helm chart / airgapped

**Describe the bug**
When we want to replicate the images based on the list of images from https://github.com/rancher/rancher/releases/download/v2.5.11/rancher-images.txt for airgapped the download of docker.io/busybox fails with "too many requests"

**To Reproduce**
Download docker.io/busybox many times on one day and you will get "too many requests"

**Result**
Airgapped images not complete

**Expected Result**
Image download works

**Additional context**

-> can we add the required image into the rancher namespace so that we do not hit the "too many requests"?

**Comment by Martin-Weiss** [comment]
This is not stale and I just created a Support Case for this..

**Comment by cbron** [comment]
SURE-5044

**Comment by KevinJoiner** [comment]
## What was fixed, or what change has occurred
- audit logging default image is now bci-micro
- bci-busybox is still included for rancher support but not required to run rancher/rancher.
- helm value of busyBoxImage is deprecated and replaced with auditLog.image
- helm value busyBoxImageTag was removed without deprecation since it was never released
- bci-micro is used for audit logging instead of bci-busybox because bci-busybox does not have the required tail option --retry needed to run the sidecar pod.
- helm value auditlog.image will be used by default unless the value busyBoxImage is set.

## Areas or cases that should be tested
Audit logging via HA install.

### Verify audit logs are still visible
1. Install rancher via helm is the following values
    ```
    helm install rancher rancher-latest/rancher \                                                             
    --namespace cattle-system \
    --set hostname=<site_url> \
    --set auditLog.level=2
    ```
2. Log Into rancher and create a new deployment of any kind `nginx:latest`
3. Verify that auditLogs are being produced
    ```
    kubectl logs -n cattle-system -f rancher-<gen_name> rancher-audit-log
    ```
4.  verify that bci-micro image is being used for the rancher sidecar. 
    ```
    kubectl get deployment -n cattle-system rancher -o yaml | grep bci-micro
    ```
   
### Users who upgrade without the busyBoxImage set should automatically be switched to using bci-micro.
1. Install rancher via helm is the following values
    ```
    helm install rancher rancher-stable/rancher \                                                             
    --namespace cattle-system \
    --set hostname=<site_url> \
    --set auditLog.level=2
    ```
    
2. Upgrade rancher via helm is the following values
    ```
    helm upgrade rancher rancher-latest/rancher \                                                             
    --namespace cattle-system \
    --set hostname=<site_url> \
    --set auditLog.level=2
    ```
2. Log Into rancher and create a new deployment of any kind `nginx:latest`
3. Verify that auditLogs are being produced
    ```
    kubectl logs -n cattle-system -f rancher-<gen_name> rancher-audit-log
    ```
4.  verify that bci-micro image is being used for the rancher sidecar. 
    ```
    kubectl get deployment -n cattle-system rancher -o yaml | grep bci-micro
    ```
   
### Users who upgrade with the busyBoxImage set should still use the image that was previously set.
1. Install rancher via helm is the following values
    ```
    helm install rancher rancher-stable/rancher \                                                             
    --namespace cattle-system \
    --set hostname=<site_url> \
    --set auditLog.level=2
    --set busyBoxImage="busybox"
    ```
    
2. Upgrade rancher via helm is the following values
    ```
    helm upgrade rancher rancher-latest/rancher \                                                             
    --namespace cattle-system \
    --set hostname=<site_url> \
    --set auditLog.level=2
    ```
2. Log Into rancher and create a new deployment of any kind `nginx:latest`
3. Verify that auditLogs are being produced
    ```
    kubectl logs -n cattle-system -f rancher-<gen_name> rancher-audit-log
    ```
4.  verify that bci-micro image is being used for the rancher sidecar. 
    ```
    kubectl get deployment -n cattle-system rancher -o yaml | grep busybox
    ```
## What areas could experience regressions
generated rancher-image.txt for the release should no longer contain busybox

## Are the repro steps accurate/minimal?
Yes

**Comment by brudnak** [comment]
# 🚧 Reopened

>See the two failures in the test table at the bottowm
>TLDR: busybox image is still showing up in the rancher-images.txt file.
>And when upgrading from 2.6.8 to 2.7.0-rc3 even with busybox set in the helm
>install and upgrade it's being changed to bci-micro.

## Reproduction Environment


| Component                        | Version / Type                           |
| -------------------------------- | ---------------------------------------- |
| Rancher version                  | 2.6.8                                    |
| Installation option              | Helm (high availability)                 |
| If Helm Chart k8s cluster        | v1.24.6+k3s1                             |
| Cert Details                     | Let's Encrypt / `traefik`                |
| k3s ha external db               | Aurora MySQL 5.7.mysql_aurora.2.10.2     |
| Helm version                     | v2.16.8-rancher2                         |
| Downstream cluster type          | Not Applicable                           |
| Downstream K8s version           | Not Applicable                           |
| Authentication providers enabled | local                                    |
| Logged in user role              | Administrator & Standard User            |
| Browser type                     | Google Chrome                            |
| Browser version                  | 106.0.5249.103 (Official Build) (x86_64) |

<details>
    <summary>🚨 Additional Reproduction Setup Details: Click to Expand</summary>

```shell
helm install rancher rancher-latest/rancher \
  --namespace cattle-system \
  --set hostname=$URL_VAR \
  --set bootstrapPassword=$PW_VAR \
  --set ingress.tls.source=letsEncrypt \
  --set letsEncrypt.email=$EMAIL_VAR \
  --set letsEncrypt.ingress.class=traefik \
  --set auditLog.level=2
```

</details>


## Reproduction steps

- Install Rancher with the following helm install command:

```shell
helm install rancher rancher-latest/rancher \
  --namespace cattle-system \
  --set hostname=$URL_VAR \
  --set bootstrapPassword=$PW_VAR \
  --set ingress.tls.source=letsEncrypt \
  --set letsEncrypt.email=$EMAIL_VAR \
  --set letsEncrypt.ingress.class=traefik \
  --set auditLog.level=2
```

- Check the logging sidecar being used with: `kubectl get deployment -n cattle-system rancher -o yaml`

```yaml
        volumeMounts:
        - mountPath: /var/log/auditlog
          name: audit-log
      - args:
        - -F
        - /var/log/auditlog/rancher-api-audit.log
        command:
        - tail
        image: busybox
        imagePullPolicy: IfNotPresent
        name: rancher-audit-log
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /var/log/auditlog
```

## Additional Info

## RESULTS

#### ✅ Expected
This is being updated to `bci-micro`

#### ❌ Actual
Since this is an older version of Rancher we still just get `busybox`

<hr>

## Validation Environment


| Component                        | Version / Type                           |
| -------------------------------- | ---------------------------------------- |
| Rancher version                  | 2.7.0-rc3                                |
| Installation option              | Helm (high availability)                 |
| If Helm Chart k8s cluster        | v1.24.6+k3s1                             |
| Cert Details                     | Let's Encrypt / `traefik`                |
| k3s ha external db               | Aurora MySQL 5.7.mysql_aurora.2.10.2     |
| Helm version                     | v2.16.8-rancher2                         |
| Downstream cluster type          | Not Applicable                           |
| Downstream K8s version           | Not Applicable                           |
| Authentication providers enabled | local & GitHub                           |
| Logged in user role              | Administrator & Standard User            |
| Browser type                     | Google Chrome                            |
| Browser version                  | 106.0.5249.103 (Official Build) (x86_64) |


<details>
    <summary>🚨 Additional Validation Setup Details: Click to Expand</summary>

```shell
  helm install rancher rancher-latest/rancher \
  --namespace cattle-system \
  --set hostname=$URL_VAR \
  --set bootstrapPassword=$PW_VAR \
  --set ingress.tls.source=letsEncrypt \
  --set letsEncrypt.email=$EMAIL_VAR \
  --set letsEncrypt.ingress.class=traefik \
  --set auditLog.level=2 \
  --version 2.7.0-rc3
```

</details>

## Validation steps

#### Scenario
**Verify audit logs are still visible on a fresh install of latest 2.7.0-rc**

1. Create a Rancher with `auditLog.level=2` set in the helm install command
2. Log Into rancher and create a new deployment of any kind nginx:latest
3. Verify that auditLogs are being produced
4. Verify that bci-micro image is being used for the rancher sidecar


## Additional Info

## RESULTS

#### ✅ Expected
Logs are produced and the sidecar for logging being used is bci-micro

#### ✅ Actual
Logs are produced and the sidecar for logging being used is bci-micro

## 📋 QA Test Plan

<table>
    <tbody>
        <tr>
            <th>Scenario</th>
            <th>Steps</th>
            <th>Expected</th>
            <th>Actual & Pass/Fail</th>
        </tr>
        <tr>
            <td>Users who upgrade without the busyBoxImage set should automatically be switched to using bci-micro.</td>
            <td>
                <ol>
                    <li>Start with a 2.6.8 Rancher server</li>
                    <li>Verify that busybox is being used</li>
                    <li>Upgrade to the latest 2.7.0-rc</li>
                    <li>Confirm that bci-micro is now being used instead of busybox</li>
                    <li>Ensure that logging is working before and after upgrade</li>
                </ol>
            </td>
            <td>Expected: For logging to work after the upgrade and for the sidecar image to be bci-micro</td>
            <td>Actual: Logging is working and the sidecar iamge is bci-micro<br><br>Pass: ✅<code>PASS</code></td>
        </tr>
        <tr>
            <td>Users who upgrade with the busyBoxImage set should still use the image that was previously set.</td>
            <td>
                <ol>
                    <li>Start with a Rancher server 2.6.8 installed with this passed into heml install <code>--set busyBoxImage="busybox"</code></li>
                    <li>While on 2.6.8 confirm busybox is being used</li>
                    <li>Upgrade Rancher to 2.7.0-rc3 using the same install command, meaning <code>--set busyBoxImage="busybox"</code> is passed again with helm upgrade</li>
                    <li>After upgrading to 2.7.0-rc3 check if busybox is still used with:</li>
                    <li><code>kubectl get deployment -n cattle-system rancher -o yaml | grep busybox</code></li>
                    <li>busybox is no longer being used</li>
                    <li>It has switched to: <code>image: registry.suse.com/bci/bci-micro:15.4.14.3</code></li>
                </ol>
            </td>
            <td>Expected: As described in the dev template, if busybox is set in the helm install command, and with the upgrade for it to remain</td>
            <td>Actual: On upgrade busybox changed to bci-micro<br><br>Fail: ❌<code>FAIL</code></td>
        </tr>
        <tr>
            <td>generated rancher-image.txt for the release should no longer contain busybox</td>
            <td>
                <ol>
                    <li>Check the rancher-images.txt file for rancher 2.7.0-rc3</li>
                </ol>
            </td>
            <td>Expected: To no contain busybox</td>
            <td>Actual: The file still contains references to busybox<br><br>Fail: ❌<code>FAIL</code></td>
        </tr>
    </tbody>
    </tbody>
</table>

**Comment by brudnak** [comment]
## Validation Retest Environment


| Component                        | Version / Type                           |
| -------------------------------- | ---------------------------------------- |
| Rancher version                  | 2.7.0-rc3                                |
| Installation option              | Helm (high availability)                 |
| If Helm Chart k8s cluster        | v1.24.6+k3s1                             |
| Cert Details                     | Let's Encrypt / `traefik`                |
| k3s ha external db               | Aurora MySQL 5.7.mysql_aurora.2.10.2     |
| Helm version                     | v2.16.8-rancher2                         |
| Downstream cluster type          | Not Applicable                           |
| Downstream K8s version           | Not Applicable                           |
| Authentication providers enabled | local & GitHub                           |
| Logged in user role              | Administrator & Standard User            |
| Browser type                     | Google Chrome                            |
| Browser version                  | 106.0.5249.103 (Official Build) (x86_64) |


<table>
    <tbody>
        <tr>
            <th>Scenario</th>
            <th>Steps</th>
            <th>Expected</th>
            <th>Actual & Pass/Fail</th>
        </tr>
        <tr>
            <td>Users who upgrade with the busyBoxImage set should still use the image that was previously set.</td>
            <td>
                <ol>
                    <li>Start with a Rancher server 2.6.8 installed with this passed into heml install <code>--set busyboxImage="busybox"</code></li>
                    <li>While on 2.6.8 confirm busybox is being used</li>
                    <li>Upgrade Rancher to 2.7.0-rc3 using the same install command, meaning <code>--set busyboxImage="busybox"</code> is passed again with helm upgrade</li>
                    <li>After upgrading to 2.7.0-rc3 check if busybox is still used with:</li>
                    <li><code>kubectl get deployment -n cattle-system rancher -o yaml | grep busybox</code></li>
                    <li>busybox is still the sidecar</li>
                    <li>Created a deployment</li>
                    <li>Verified that logging is still working</li>
                </ol>
            </td>
            <td>Expected: As described in the dev template, if busybox is set in the helm install command, and with the upgrade for it to remain</td>
            <td>Actual: busybox was still the image in use, and logging was working<br><br>Pass/Fail: ✅<code>PASS</code></td>
        </tr>
        <tr>
            <td>generated rancher-image.txt for the release should no longer contain busybox</td>
            <td>
                <ol>
                    <li>Check the rancher-images.txt file for rancher 2.7.0-rc3</li>
                </ol>
            </td>
            <td>Expected: for the rancher-images.txt to only contain mirrored version of busybox</td>
            <td>Actual: rancher-images.txt only has the mirrored versions of busybox<br><br>Pass/Fail: ✅<code>PASS</code></td>
        </tr>
    </tbody>
    </tbody>
</table>

**Comment by KevinJoiner** [comment]
# Release Notes
## Synopsis of issue
Rancher now defaults to using the bci-micro image for its sidecare audit logging instead of busybox.
## Resolved (or not):
Resolved.
## Status: 
Resolved in v2.7.0
## Versions: 
< v2.7.0
